controllers/gitrepository_controller.go

```diff
@@ -39,6 +39,10 @@ import (
 	intgit "github.com/fluxcd/source-controller/internal/git"
 )
 
+const (
+	defaultGitTimeout = time.Second * 15
```

Comment:
I would move this to the API package and change it to 20s to match Flux. When we'll be using CRD v1, we can set the default in the CRD object see https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#defaulting